Revised notebook to demonstrate remapping MCS mask from lat/lon grid to healpix

feng045 opened the pull request:
This is revised from PR #16. Thanks @markmuetz and @d70-t for the inputs and suggestions!

1. Changed reading HEALPix data from a catalog.
2. Simplified the function to fix coordinates of the MCS mask dataset on lat/lon grid.
3. Updated remap to HEALPix method that works for non-global grid.
4. Reduced plot sizes and notebook file size.

markmuetz reviewed (approved):
Looks good to me. You could set DPI at the top of the doc (it's missing from first plot):

```python
import matplotlib as mpl
mpl.rcParams['figure.dpi'] = 75

```

You could also add tracking masks to the NERSC catalog - this would be nice.

I would probably make `tolerance` an arg to the `is_valid` function with default value 0.1. 

I think the `fix_coords` is slightly different from the one suggested by @d70-t , yours changes the overall layout of the data such that the 0-360 data goes from 0 to 360 (instead of from 180 to 360, then 0 to 180 as per @d70-t 's). I need this for my code - will borrow.

But these are minor nitpicks. Happy with the logic and file size seems OK to me.

d70-t reviewed (commented):
Thanks for revising the notebook. It's already much more concise 🎉 

* I'd suggest to have a look at #50, which loads *one* of the sub-catalogs first, to improve code-protabiltiy. If we agree on that approach, this notebook should follow the same approach.
* the variables `dkm` and `res` are not used
* `dask.array` is not used
* should the MCS mask eventually be part of the catalog?
* the notebook contains a couple of cells which seem like remainders of testing, maybe you could have skim over them again
* `lon_hp` and `lat_hp` probably should correspond to the coordinate system of the model output you want to compare to (i.e. the SCREAM output). Why don't you just take the coordinates of that dataset (probably using `ds=ds.pipe(egh.attach_coords)` as in [the simple plot example](https://github.com/digital-earths-global-hackathon/hk25-teams/blob/main/hk25-tutorials/simple_plot.ipynb), also discussed in #50

feng045 commented:
Thanks for the comments @markmuetz and @d70-t!  I further revised the notebook following your suggestions:
1. Updated catalog from hackathon Github following #50.
2. Used `lon` and `lat` coordinates from the model HEALPix data by `ds_hp = ds_hp.pipe(egh.attach_coords)`.
3. Updated `is_valid` function with tolerance argument.
4. Further clean up.

Regarding rolling the `lon` coordinate in the `fix_coords` function, I tried a couple of things:

- Not changing the `lon` coordinate from (-180 to + 180) to (0 to 360). This doesn't work, as only half of the domain (0 to +180) was remapped (see attached plots).
- Not rolling the `lon` dimension after converting to (0 to 360), this results in an error complaining the index was not monotonically increasing.

In my case where the original data is (-180 to + 180), I think rolling is necessary.  So I kept the `fix_coords` function as is.
![mask_latlon](https://github.com/user-attachments/assets/6568dfa3-6a55-49b8-93af-ae8a8b878697)
![mask_healpix](https://github.com/user-attachments/assets/a946022d-2dc3-4042-8be9-f5fb7608d37c)

markmuetz reviewed (approved):
All looks good to me now!

d70-t commented:
> Not changing the lon coordinate from (-180 to + 180) to (0 to 360). This doesn't work, as only half of the domain (0 to +180) was remapped (see attached plots).

Did you also map the coordinates from the healpix dataset to the -180...180 range before selecting from the original data? E.g.:
* by applying `((lon + 180) % 360) - 180)`
* or by using the [`signed=True`](https://github.com/mpimet/easygems/blob/946cf64c37476851b5dc4ac2d6e6819f67c3a7be/easygems/healpix/__init__.py#L67) argument of `attach_coords` ?

feng045 commented:
Thanks @d70-t for another neat trick with [`signed=True`](https://github.com/mpimet/easygems/blob/946cf64c37476851b5dc4ac2d6e6819f67c3a7be/easygems/healpix/__init__.py#L67) argument of `attach_coords`!

I tested it and it works without rolling the input dataset, further simplifying the code and reducing the computation time by ~30% based on testing the conversion for a full year of hourly data at zoom level 9!  The update is now in #57.
